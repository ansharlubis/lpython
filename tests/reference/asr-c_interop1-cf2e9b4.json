{
    "basename": "asr-c_interop1-cf2e9b4",
    "cmd": "lpython --show-asr --no-color {infile} -o {outfile}",
    "infile": "tests/c_interop1.py",
    "infile_hash": "e676d05044fa71c2bd65b09dd5ed2d20414f6b1c028ef528aab47748",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr-c_interop1-cf2e9b4.stdout",
    "stdout_hash": "26f1834c86739c0ab7ec832996e8a5119108e4ef89aa30b5a314283b",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}