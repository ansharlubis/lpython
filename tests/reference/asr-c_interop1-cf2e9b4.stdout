([1m[35mTranslationUnit[39m[0m 
   ([33mSymbolTable[39m 
      1 
      {
         f: 
            ([1m[35mFunction[39m[0m 
               ([33mSymbolTable[39m 
                  2 
                  {
                     _lpython_return_variable: 
                        ([1m[35mVariable[39m[0m 
                           2 
                           _lpython_return_variable [1m[32m
                           ReturnVar[39m[0m () () [1m[32m
                           Default[39m[0m 
                           ([1m[35mReal[39m[0m 8 []) [1m[32m
                           BindC[39m[0m [1m[32m
                           Public[39m[0m [1m[32m
                           Required[39m[0m .false.), 
                     x: 
                        ([1m[35mVariable[39m[0m 
                           2 
                           x [1m[32m
                           In[39m[0m () () [1m[32m
                           Default[39m[0m 
                           ([1m[35mReal[39m[0m 8 []) [1m[32m
                           BindC[39m[0m [1m[32m
                           Public[39m[0m [1m[32m
                           Required[39m[0m .true.)
                  }) 
               f [
               ([1m[35mVar[39m[0m 2 [33mx[39m)] [] 
               ([1m[35mVar[39m[0m 2 [33m_lpython_return_variable[39m) [1m[32m
               BindC[39m[0m [1m[32m
               Public[39m[0m [1m[32m
               Interface[39m[0m ()), 
         g: 
            ([1m[35mSubroutine[39m[0m 
               ([33mSymbolTable[39m 
                  3 
                  {
                     a: 
                        ([1m[35mVariable[39m[0m 
                           3 
                           a [1m[32m
                           In[39m[0m () () [1m[32m
                           Default[39m[0m 
                           ([1m[35mReal[39m[0m 8 []) [1m[32m
                           BindC[39m[0m [1m[32m
                           Public[39m[0m [1m[32m
                           Required[39m[0m .true.), 
                     b: 
                        ([1m[35mVariable[39m[0m 
                           3 
                           b [1m[32m
                           In[39m[0m () () [1m[32m
                           Default[39m[0m 
                           ([1m[35mReal[39m[0m 4 []) [1m[32m
                           BindC[39m[0m [1m[32m
                           Public[39m[0m [1m[32m
                           Required[39m[0m .true.), 
                     c: 
                        ([1m[35mVariable[39m[0m 
                           3 
                           c [1m[32m
                           In[39m[0m () () [1m[32m
                           Default[39m[0m 
                           ([1m[35mInteger[39m[0m 8 []) [1m[32m
                           BindC[39m[0m [1m[32m
                           Public[39m[0m [1m[32m
                           Required[39m[0m .true.), 
                     d: 
                        ([1m[35mVariable[39m[0m 
                           3 
                           d [1m[32m
                           In[39m[0m () () [1m[32m
                           Default[39m[0m 
                           ([1m[35mInteger[39m[0m 4 []) [1m[32m
                           BindC[39m[0m [1m[32m
                           Public[39m[0m [1m[32m
                           Required[39m[0m .true.)
                  }) 
               g [
               ([1m[35mVar[39m[0m 3 [33ma[39m) 
               ([1m[35mVar[39m[0m 3 [33mb[39m) 
               ([1m[35mVar[39m[0m 3 [33mc[39m) 
               ([1m[35mVar[39m[0m 3 [33md[39m)] [] [1m[32m
               BindC[39m[0m [1m[32m
               Public[39m[0m [1m[32m
               Interface[39m[0m () .false. .false.), 
         h: 
            ([1m[35mFunction[39m[0m 
               ([33mSymbolTable[39m 
                  4 
                  {
                     _lpython_return_variable: 
                        ([1m[35mVariable[39m[0m 
                           4 
                           _lpython_return_variable [1m[32m
                           ReturnVar[39m[0m () () [1m[32m
                           Default[39m[0m 
                           ([1m[35mReal[39m[0m 8 []) [1m[32m
                           BindC[39m[0m [1m[32m
                           Public[39m[0m [1m[32m
                           Required[39m[0m .false.), 
                     x: 
                        ([1m[35mVariable[39m[0m 
                           4 
                           x [1m[32m
                           In[39m[0m () () [1m[32m
                           Default[39m[0m 
                           ([1m[35mReal[39m[0m 8 []) [1m[32m
                           BindC[39m[0m [1m[32m
                           Public[39m[0m [1m[32m
                           Required[39m[0m .true.)
                  }) 
               h [
               ([1m[35mVar[39m[0m 4 [33mx[39m)] [
               ([1m[35m=[39m[0m 
                  ([1m[35mVar[39m[0m 4 [33m_lpython_return_variable[39m) 
                  ([1m[35mRealBinOp[39m[0m 
                     ([1m[35mVar[39m[0m 4 [33mx[39m) [1m[32m
                     Add[39m[0m 
                     ([1m[35mRealConstant[39m[0m   1.00000000000000000e+00 
                        ([1m[35mReal[39m[0m 8 [])) 
                     ([1m[35mReal[39m[0m 8 []) ()) ()) 
               ([1m[35mReturn[39m[0m)] 
               ([1m[35mVar[39m[0m 4 [33m_lpython_return_variable[39m) [1m[32m
               BindC[39m[0m [1m[32m
               Public[39m[0m [1m[32m
               Implementation[39m[0m ()), 
         l: 
            ([1m[35mSubroutine[39m[0m 
               ([33mSymbolTable[39m 
                  5 
                  {
                     a: 
                        ([1m[35mVariable[39m[0m 
                           5 
                           a [1m[32m
                           In[39m[0m () () [1m[32m
                           Default[39m[0m 
                           ([1m[35mReal[39m[0m 8 []) [1m[32m
                           BindC[39m[0m [1m[32m
                           Public[39m[0m [1m[32m
                           Required[39m[0m .true.), 
                     b: 
                        ([1m[35mVariable[39m[0m 
                           5 
                           b [1m[32m
                           In[39m[0m () () [1m[32m
                           Default[39m[0m 
                           ([1m[35mReal[39m[0m 4 []) [1m[32m
                           BindC[39m[0m [1m[32m
                           Public[39m[0m [1m[32m
                           Required[39m[0m .true.), 
                     c: 
                        ([1m[35mVariable[39m[0m 
                           5 
                           c [1m[32m
                           In[39m[0m () () [1m[32m
                           Default[39m[0m 
                           ([1m[35mInteger[39m[0m 8 []) [1m[32m
                           BindC[39m[0m [1m[32m
                           Public[39m[0m [1m[32m
                           Required[39m[0m .true.), 
                     d: 
                        ([1m[35mVariable[39m[0m 
                           5 
                           d [1m[32m
                           In[39m[0m () () [1m[32m
                           Default[39m[0m 
                           ([1m[35mInteger[39m[0m 4 []) [1m[32m
                           BindC[39m[0m [1m[32m
                           Public[39m[0m [1m[32m
                           Required[39m[0m .true.)
                  }) 
               l [
               ([1m[35mVar[39m[0m 5 [33ma[39m) 
               ([1m[35mVar[39m[0m 5 [33mb[39m) 
               ([1m[35mVar[39m[0m 5 [33mc[39m) 
               ([1m[35mVar[39m[0m 5 [33md[39m)] [
               ([1m[35mPrint[39m[0m () [
                  ([1m[35mStringConstant[39m[0m "OK" 
                     ([1m[35mCharacter[39m[0m 1 2 () []))] () ())] [1m[32m
               BindC[39m[0m [1m[32m
               Public[39m[0m [1m[32m
               Implementation[39m[0m () .false. .false.), 
         main0: 
            ([1m[35mSubroutine[39m[0m 
               ([33mSymbolTable[39m 
                  6 
                  {
                     i: 
                        ([1m[35mVariable[39m[0m 
                           6 
                           i [1m[32m
                           Local[39m[0m () () [1m[32m
                           Default[39m[0m 
                           ([1m[35mReal[39m[0m 8 []) [1m[32m
                           Source[39m[0m [1m[32m
                           Public[39m[0m [1m[32m
                           Required[39m[0m .false.), 
                     x: 
                        ([1m[35mVariable[39m[0m 
                           6 
                           x [1m[32m
                           Local[39m[0m () () [1m[32m
                           Default[39m[0m 
                           ([1m[35mReal[39m[0m 8 []) [1m[32m
                           Source[39m[0m [1m[32m
                           Public[39m[0m [1m[32m
                           Required[39m[0m .false.), 
                     y: 
                        ([1m[35mVariable[39m[0m 
                           6 
                           y [1m[32m
                           Local[39m[0m () () [1m[32m
                           Default[39m[0m 
                           ([1m[35mReal[39m[0m 4 []) [1m[32m
                           Source[39m[0m [1m[32m
                           Public[39m[0m [1m[32m
                           Required[39m[0m .false.), 
                     z: 
                        ([1m[35mVariable[39m[0m 
                           6 
                           z [1m[32m
                           Local[39m[0m () () [1m[32m
                           Default[39m[0m 
                           ([1m[35mInteger[39m[0m 8 []) [1m[32m
                           Source[39m[0m [1m[32m
                           Public[39m[0m [1m[32m
                           Required[39m[0m .false.), 
                     zz: 
                        ([1m[35mVariable[39m[0m 
                           6 
                           zz [1m[32m
                           Local[39m[0m () () [1m[32m
                           Default[39m[0m 
                           ([1m[35mInteger[39m[0m 4 []) [1m[32m
                           Source[39m[0m [1m[32m
                           Public[39m[0m [1m[32m
                           Required[39m[0m .false.)
                  }) 
               main0 [] [
               ([1m[35m=[39m[0m 
                  ([1m[35mVar[39m[0m 6 [33mx[39m) 
                  ([1m[35mRealConstant[39m[0m   5.00000000000000000e+00 
                     ([1m[35mReal[39m[0m 8 [])) ()) 
               ([1m[35m=[39m[0m 
                  ([1m[35mVar[39m[0m 6 [33mi[39m) 
                  ([1m[35mFunctionCall[39m[0m 1 [33mf[39m () [
                     (
                        ([1m[35mVar[39m[0m 6 [33mx[39m))] 
                     ([1m[35mReal[39m[0m 8 []) () ()) ()) 
               ([1m[35m=[39m[0m 
                  ([1m[35mVar[39m[0m 6 [33my[39m) 
                  ([1m[35mCast[39m[0m 
                     ([1m[35mRealConstant[39m[0m   5.40000000000000036e+00 
                        ([1m[35mReal[39m[0m 8 [])) [1m[32m
                     RealToReal[39m[0m 
                     ([1m[35mReal[39m[0m 4 []) 
                     ([1m[35mRealConstant[39m[0m   5.40000000000000036e+00 
                        ([1m[35mReal[39m[0m 4 []))) ()) 
               ([1m[35m=[39m[0m 
                  ([1m[35mVar[39m[0m 6 [33mz[39m) 
                  ([1m[35mCast[39m[0m ([1m[35mIntegerConstant[39m[0m [36m3[39m 
                     ([1m[35mInteger[39m[0m 4 [])) [1m[32m
                     IntegerToInteger[39m[0m 
                     ([1m[35mInteger[39m[0m 8 []) ()) ()) 
               ([1m[35m=[39m[0m 
                  ([1m[35mVar[39m[0m 6 [33mzz[39m) ([1m[35mIntegerConstant[39m[0m [36m2[39m 
                  ([1m[35mInteger[39m[0m 4 [])) ()) 
               ([1m[35mSubroutineCall[39m[0m 1 [33mg[39m () [
                  (
                     ([1m[35mVar[39m[0m 6 [33mx[39m)) 
                  (
                     ([1m[35mVar[39m[0m 6 [33my[39m)) 
                  (
                     ([1m[35mVar[39m[0m 6 [33mz[39m)) 
                  (
                     ([1m[35mVar[39m[0m 6 [33mzz[39m))] ()) 
               ([1m[35m=[39m[0m 
                  ([1m[35mVar[39m[0m 6 [33mi[39m) 
                  ([1m[35mFunctionCall[39m[0m 1 [33mh[39m () [
                     (
                        ([1m[35mVar[39m[0m 6 [33mx[39m))] 
                     ([1m[35mReal[39m[0m 8 []) () ()) ()) 
               ([1m[35mSubroutineCall[39m[0m 1 [33ml[39m () [
                  (
                     ([1m[35mVar[39m[0m 6 [33mx[39m)) 
                  (
                     ([1m[35mVar[39m[0m 6 [33my[39m)) 
                  (
                     ([1m[35mVar[39m[0m 6 [33mz[39m)) 
                  (
                     ([1m[35mVar[39m[0m 6 [33mzz[39m))] ())] [1m[32m
               Source[39m[0m [1m[32m
               Public[39m[0m [1m[32m
               Implementation[39m[0m () .false. .false.)
      }) 
   [])
(TranslationUnit (SymbolTable 1 {f: (Function (SymbolTable 2 {_lpython_return_variable: (Variable 2 _lpython_return_variable ReturnVar () () Default (Real 8 []) BindC Public Required .false.), x: (Variable 2 x In () () Default (Real 8 []) BindC Public Required .true.)}) f [(Var 2 x)] [] (Var 2 _lpython_return_variable) BindC Public Interface ()), g: (Subroutine (SymbolTable 3 {a: (Variable 3 a In () () Default (Real 8 []) BindC Public Required .true.), b: (Variable 3 b In () () Default (Real 4 []) BindC Public Required .true.), c: (Variable 3 c In () () Default (Integer 8 []) BindC Public Required .true.), d: (Variable 3 d In () () Default (Integer 4 []) BindC Public Required .true.)}) g [(Var 3 a) (Var 3 b) (Var 3 c) (Var 3 d)] [] BindC Public Interface () .false. .false.), h: (Function (SymbolTable 4 {_lpython_return_variable: (Variable 4 _lpython_return_variable ReturnVar () () Default (Real 8 []) BindC Public Required .false.), x: (Variable 4 x In () () Default (Real 8 []) BindC Public Required .true.)}) h [(Var 4 x)] [(= (Var 4 _lpython_return_variable) (RealBinOp (Var 4 x) Add (RealConstant   1.00000000000000000e+00 (Real 8 [])) (Real 8 []) ()) ()) (Return)] (Var 4 _lpython_return_variable) BindC Public Implementation ()), l: (Subroutine (SymbolTable 5 {a: (Variable 5 a In () () Default (Real 8 []) BindC Public Required .true.), b: (Variable 5 b In () () Default (Real 4 []) BindC Public Required .true.), c: (Variable 5 c In () () Default (Integer 8 []) BindC Public Required .true.), d: (Variable 5 d In () () Default (Integer 4 []) BindC Public Required .true.)}) l [(Var 5 a) (Var 5 b) (Var 5 c) (Var 5 d)] [(Print () [(StringConstant "OK" (Character 1 2 () []))] () ())] BindC Public Implementation () .false. .false.), main0: (Subroutine (SymbolTable 6 {i: (Variable 6 i Local () () Default (Real 8 []) Source Public Required .false.), x: (Variable 6 x Local () () Default (Real 8 []) Source Public Required .false.), y: (Variable 6 y Local () () Default (Real 4 []) Source Public Required .false.), z: (Variable 6 z Local () () Default (Integer 8 []) Source Public Required .false.), zz: (Variable 6 zz Local () () Default (Integer 4 []) Source Public Required .false.)}) main0 [] [(= (Var 6 x) (RealConstant   5.00000000000000000e+00 (Real 8 [])) ()) (= (Var 6 i) (FunctionCall 1 f () [((Var 6 x))] (Real 8 []) () ()) ()) (= (Var 6 y) (Cast (RealConstant   5.40000000000000036e+00 (Real 8 [])) RealToReal (Real 4 []) (RealConstant   5.40000000000000036e+00 (Real 4 []))) ()) (= (Var 6 z) (Cast (IntegerConstant 3 (Integer 4 [])) IntegerToInteger (Integer 8 []) ()) ()) (= (Var 6 zz) (IntegerConstant 2 (Integer 4 [])) ()) (SubroutineCall 1 g () [((Var 6 x)) ((Var 6 y)) ((Var 6 z)) ((Var 6 zz))] ()) (= (Var 6 i) (FunctionCall 1 h () [((Var 6 x))] (Real 8 []) () ()) ()) (SubroutineCall 1 l () [((Var 6 x)) ((Var 6 y)) ((Var 6 z)) ((Var 6 zz))] ())] Source Public Implementation () .false. .false.), main_program: (Program (SymbolTable 7 {}) main_program [] [])}) [])
